tools/install.sh

```diff
@@ -1,23 +1,47 @@
 #!/bin/bash
+set -o pipefail
 
 repository="davrodpin/mole"
 install_path="/usr/local/bin"
+temporary_file="/tmp/mole.tar.gz"
 
-latest_version=$(curl --silent "https://api.github.com/repos/${repository}/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+# Get the os architecture
+os_arch=$(uname -m)
 
-# Get the os type
-os_type=$(uname -sm)
+# Check if architecture is supported
+if [ "${os_arch,,}" != "x86_64" ]; then
+	echo "The ${os_arch} architecture is not supported"
+	exit 1
+fi
 
-# Extract the first part of os_type (Linux/Darwin)
-os_type=${os_type% *}
+# Get the OS type
+os_type=$(uname -s)
 
 # Convert os_type to lowercase
 os_type=${os_type,,}
 
-filename="mole${latest_version#v}.${os_type}-amd64.tar.gz"
-
-download_link="https://github.com/davrodpin/mole/releases/download/${latest_version}/${filename}"
+# Get latest version of mole available
+latest_version=$(curl --silent --location --max-time 60 "https://api.github.com/repos/${repository}/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+if [ $? -ne 0 ]; then
+	echo -ne "\nThere was an error trying to check what is the latest version of mole.\nPlease try again later.\n"
+	exit 1
+fi
 
-curl -L "${download_link}" | sudo tar xz -C "${install_path}"
+filename="mole${latest_version#v}.${os_type}-amd64.tar.gz"
+download_link="https://github.com/${repository}/releases/download/${latest_version}/${filename}"
+
+# Download latest version of mole available
+curl --location --max-time 60 "${download_link}" -o "${temporary_file}"
```

Comment:
I like your previous code that is less dependent on the file system (`/tmp`).

You could keep that and handle the error, something like:

```sh
$ curl --location --max-time 60 "${download_link}" | sudo tar -xzf -C "${install_path}" || {
  echo -ne "..."
}
```